Lib/sysconfig.py

```diff
@@ -696,6 +699,10 @@ def get_config_vars(*args):
             import _osx_support
             _osx_support.customize_config_vars(_CONFIG_VARS)
 
+    with _CONFIG_VARS_LOCK:
+        if _CONFIG_VARS is None:
+            init_config_vars()
+
```

Comment:
It's a bit of a shame that every read now needs to get a lock, otherwise the code looks good to me.

Reply:
I had a similar thought, but I wanted to keep my change short and easy to review.

The "run some code at most once, so that only the first caller, and any callers racing with it, have to take a lock" problem occurs in lots of places and so I think the proper way to solve it would be to add a new primitive in the `threading` module, corresponding to [`pthread_once`](https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_once.html) in POSIX threads, or [`InitOnceExecuteOnce`](https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-initonceexecuteonce) in Windows. But that's well beyond the scope of this PR.

Reply:
What about something like this?

```suggestion
    if _CONFIG_VARS is None:  # avoid checking the lock for other cases
        with _CONFIG_VARS_LOCK:
            if _CONFIG_VARS is None:  # check again, with the lock
                init_config_vars()

```

Reply:
That re-introduces the race condition, since `_CONFIG_VARS` is assigned early in `init_config_vars()`. However, I can re-write to postpone the assignment to `_CONFIG_VARS` (unfortunately that makes the diff much noisier).

Reply:
As you can see from the test failures, this won't work because of the re-entrancy in sysconfig.py. `get_config_vars()` calls `_init_posix()` or `_init_non_posix()`, which both call `get_path()`, which calls `get_paths()`, which calls `_expand_vars()`, which calls `get_config_vars()`. So we can't use `_CONFIG_VARS is None` as our guard for the initialization code.

Reply:
> That re-introduces the race condition...

My idea was that in that case, it goes into the if-block unnecessarily, but then the lock stops it from going further, it waits until the initialization is complete and does nothing because the second check will be False.

Reply:
Never mind me, the first check will be False, that was your point, I go it now.

Reply:
I pushed a fixup commit that adds a global `_CONFIG_VARS_INITIALIZED` and otherwise is the same as your suggestion.

Reply:
This now looks good to me.

Consider moving the comments above the lines so the lines are still [reasonably short](https://peps.python.org/pep-0008/#maximum-line-length).

Reply:
Done.